### PR TITLE
More details when generating inbounds

### LINKED ISSUE #320: Inbound Shipment fixes
Splitting off from #225 

**3. KHC receives an Inbound Shipment**
- [x] Inbound Shipment > Details Panel: shipment level comment entered by OHC is lost as well as the transport reference (could be useful for the customer cause it could be a tracking reference)
- [ ] Inbound Shipment > Details Panel > Related Documents: when I click on the link to the Req, most of the time the requisition fails to load or it takes a long time.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #320 

Populate comment, reference & surface the transport reference in the UI for inbounds.
Couldn't recreate the other issue 🤷

### CODE REVIEW

File: client/packages/common/src/ui/components/inputs/Autocomplete/AutocompleteList.tsx [hunk 1]
```diff
@@ -80,14 +80,16 @@ export const AutocompleteList = <T,>({
...
       sx={{
-        '& .MuiAutocomplete-listbox': {
-          minHeight: height ? `${height}` : 'auto',
-          maxHeight: height ? `${height}` : 'auto',
-        },
         '& .MuiAutocomplete-inputRoot': {
           width: width ? `${width}px` : 'auto',
         },
       }}
+      ListboxProps={{
```
Comment (mark-prins) [inline client/packages/common/src/ui/components/inputs/Autocomplete/AutocompleteList.tsx:87]: offroad: the customer select modal was looking like this:

Screen Shot 2022-07-26 at 11 38 35 AM

there's a long list of customers, with half of them hidden..

File: client/packages/invoices/src/InboundShipment/DetailView/SidePanel/SidePanel.tsx [hunk 2]
```diff
@@ -36,6 +37,7 @@ export const SidePanel: FC = () => {
       <AdditionalInfoSection />
       <RelatedDocumentsSection />
       <PricingSection />
+      {isTransfer && <TransportSection />}
```
Comment (mark-prins) [inline client/packages/invoices/src/InboundShipment/DetailView/SidePanel/SidePanel.tsx:40]: decided that there's no point showing this for manual shipments.. the transport reference isn't editable, so will never be populated

File: client/packages/invoices/src/InboundShipment/DetailView/SidePanel/TransportSection.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      <Grid container gap={0.5} key="transport-details">
+        <PanelLabel display="flex" alignItems="center">
+          {t('heading.reference')}
+        </PanelLabel>
+        <BufferedTextInput
+          disabled={true}
+          onChange={e => {
+            setReferenceBuffer(e.target.value);
+            update({ transportReference: e.target.value });
+          }}
```
Comment (clemens-msupply) [inline client/packages/invoices/src/InboundShipment/DetailView/SidePanel/TransportSection.tsx:36]: Could you please explain to me why this is needed/ why you do it like this? e.g. why is setReferenceBuffer called in onChange? and why call update()? isn't the e.target.value always the same as what you put into the text input?

**Review by clemens-msupply — APPROVED** [review]
LGTM! just one question about some code I don't understand....

File: client/packages/invoices/src/InboundShipment/DetailView/SidePanel/TransportSection.tsx (hunk 3 above)
Comment (mark-prins) [reply client/packages/invoices/src/InboundShipment/DetailView/SidePanel/TransportSection.tsx:36]: I believe the issue is with managing updates to the document, in this case the invoice. The problem is, if you are using only the internal state of the text component and there is an issue with the async update of the document then the value shown in the textfield won't reflect what is actually stored in the API - if the API update fails, then the API will have `value_a` while the component displays `value_b`
An alternative approach is to set the component to be read only when updating ( painful to use though when you are trying to have the component update async with a debounce callback ) or to have the value driven only by the external value ( which requires constant updating ), or to only update externally when the component loses focus ( increases the potential for the entered value to not be saved, or for dependent actions to not fire when expected, such as form validation ).

**Pull request merged**